### PR TITLE
Add basic example of wac CLI and programatic API

### LINKED ISSUE #51: Hello, world! Style example
It would be nice to have some examples. For starters, we could add a "Hello, world!" style example. I took the liberty of creating one in my own repo for now: https://github.com/rylev/wac-example. If you like it, we can move that into the `wac` repo.

### PR DESCRIPTION
**Pull request by rylev**
Fixes #51 

This adds a simple example showing how `wac encode`, `wac plug`, and the `wac-graph` API work.

### CODE REVIEW

**Comment by FrankReh** [comment]
Would there be any value in the README for this example also saying why `wac plug` could or could not be used to compose also? 

I don't think it's confusing to see what the options are. On the contrary, I'm confused when I think there are other options that aren't mentioned.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+The resulting composed component will therefore only have the exported `greet` function originally from the `greeter` component.
+
+## CLI
+
+`wac` can be used as a CLI tool. The `wac encode` command takes a wac script as input which defines how two components are composed together.
+
+Running the following command should produce a new component that is the composition of the `hello` and `greeter` components.
+
+```
+wac encode composition.wac -o composed.wasm
```
Comment (fibonacci1729) [inline examples/README.md:44]:
```suggestion
wac encode script.wac -o composed.wasm  
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+`wac` can be used as a CLI tool. The `wac encode` command takes a wac script as input which defines how two components are composed together.
+
+Running the following command should produce a new component that is the composition of the `hello` and `greeter` components.
+
+```
+wac encode composition.wac -o composed.wasm  
+```
+
+*Note*: `wac encode` expects to find any input components inside of a `deps` folder in a directory named after the namespace part of the input component's name. In our example, the wac script uses the `example:greeter` and `example:hello` input components so `wac encode` expects to find those components in the `deps/example` directory.
```
Comment (fibonacci1729) [inline examples/README.md:47]: Is it worth mentioning here in the note that where `wac` looks for `deps/` is configurable through the `--deps-dir` flag?

**Review by fibonacci1729 — APPROVED** [review]
Small fix for the name of the `wac` document used and optional clarifying question.

Looks great!

**Comment by rylev** [comment]
I tried to address all the feedback here. Let me know what you think!

**Review by fibonacci1729 — APPROVED** [review]

**Pull request merged**